### PR TITLE
Switch to Google Secret Manager for function secrets

### PR DESCRIPTION
**Pull request by mqp**
***Deploy status: Secrets are moved over on dev and prod. Functions redeployed on dev but not prod.***

This is important because v2 Cloud Functions doesn't support our old config stuff. So we need to do this switch before we can make any of our functions be v2.

### CODE REVIEW

File: functions/src/send-email.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import * as mailgun from 'mailgun-js'
-import * as functions from 'firebase-functions'
 
 const DOMAIN = 'mg.manifold.markets'
-const mg = mailgun({ apiKey: functions.config().mailgun.key, domain: DOMAIN })
 
 export const sendTextEmail = (to: string, subject: string, text: string) => {
+  const mg = mailgun({
```
Comment (jahooma) [inline functions/src/send-email.ts:6]: Would it be better to move this to a function, e.g. `initMailgun()` and then reuse below?

File: functions/src/stripe.ts [hunk 2]
```diff
@@ -42,8 +37,13 @@ const manticDollarStripePrice = isProd
     }
 
 export const createCheckoutSession = functions
-  .runWith({ minInstances: 1 })
+  .runWith({ minInstances: 1, secrets: ['STRIPE_APIKEY'] })
   .https.onRequest(async (req, res) => {
+    const stripe = new Stripe(process.env.STRIPE_APIKEY as string, {
+      apiVersion: '2020-08-27',
+      typescript: true,
+    })
```
Comment (jahooma) [inline functions/src/stripe.ts:45]: Ditto, could be better to move to function `initStripe()`

**Review by jahooma — APPROVED** [review]
Sweet! Could you update the README on the new process for adding / reading function secrets?

**Comment by mqp** [comment]
> Sweet! Could you update the README on the new process for adding / reading function secrets?

Yes, I also need to verify that it works in emulators. According to the documentation the emulators should just be able to talk to the cloud secrets no problem.

File: functions/src/send-email.ts (hunk 1 above)
Comment (mqp) [reply functions/src/send-email.ts:6]: OK, I buy that.

**Comment by mqp** [comment]
Seems to work on the emulators, so everything is good to go. Deploying on prod.

**Pull request merged**